torch_xla/csrc/ops/expand_symint.cpp

```diff
@@ -0,0 +1,72 @@
+#include "torch_xla/csrc/ops/expand_symint.h"
+
+#include "absl/strings/str_join.h"
+#include "tensorflow/compiler/xla/client/lib/constants.h"
+#include "tensorflow/compiler/xla/xla_client/debug_macros.h"
+#include "torch/csrc/lazy/core/helpers.h"
+#include "torch/csrc/lazy/core/util.h"
+#include "torch_xla/csrc/data_ops.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
+
+namespace torch_xla {
+namespace {
+
+xla::Shape NodeOutputShape(const torch::lazy::Value& input,
+                           const std::vector<int64_t> upper_bounds,
+                           const std::vector<bool> dynamic_dims) {
+  return xla::ShapeUtil::MakeShape(GetXlaShape(input).element_type(),
+                                   {upper_bounds}, {dynamic_dims});
+}
+
+std::vector<torch::lazy::Value> GetValues(
+    const torch::lazy::Value& input,
+    const std::vector<torch::lazy::NodePtr>& dimensions) {
+  std::vector<torch::lazy::Value> values;
+  values.reserve(dimensions.size() + 1);
+  values.push_back(input);
+  for (torch::lazy::NodePtr dim : dimensions) {
+    if (dim) {
+      // Dimension Node only exist for symbolic dimension.
```

Comment:
@JackCaoG is this due a different PR you recently pushed while I was away?

Reply:
I would recommend checking https://github.com/pytorch/xla/pulls?q=is%3Apr+author%3AJackCaoG+label%3Adynamism . It is one of those pr I added to refactor the symintelements.

Reply:
I see this [PR](https://github.com/pytorch/xla/pull/3944) covering the topic. Thanks.